src/yunohost/certificate.py

```diff
@@ -837,7 +843,10 @@ def _dns_ip_match_public_ip(public_ip, domain):
 
 def _domain_is_accessible_through_HTTP(ip, domain):
     try:
-        requests.head("http://" + ip, headers={"Host": domain})
+        requests.head("http://" + ip, headers={"Host": domain}, timeout=30)
+    except Timeout as e:
+        # XXX what should we do here? retry?
```

Comment:
I may not see the big picture on this, but I would go for something way more simple : if timeout after 5 to 10 seconds, then just abort with a good error message, like "timeout reached when your server tried to contact itself via its public IP address. you may very well be experiencing hairpinning, or the firewall/router ahead of your server is misconfigured"